pkg/agent/multicast/mcast_controller_test.go

```diff
@@ -1265,7 +1265,44 @@ func TestFlexibleIPAMModeInitialize(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestMulticastControllerOnIPv6Cluster(t *testing.T) {
+	for _, tc := range []struct {
+		name        string
+		ipv4Enabled bool
+		ipv6Enabled bool
+		expErr      error
+	}{
+		{
+			name:        "Failed on IPv6 only cluster",
+			ipv4Enabled: false,
+			ipv6Enabled: true,
+			expErr:      fmt.Errorf("multicast is not supported on the IPv6-only cluster"),
```

Comment:
IMO, it's better to use a string and use the `assert.EqualError` assertion. There is no need to build an error.